app/lib/backend_api_logic/routing_policy.rb

```diff
@@ -17,7 +17,7 @@ def initialize(service)
       end
 
       def to_a
-        rules = backend_api_configs.reorder(path: :desc).each_with_object([]) do |config, rules|
+        rules = backend_api_configs.reorder('CASE WHEN path IS NULL THEN 1 ELSE 0 END, path DESC').each_with_object([]) do |config, rules|
```

Comment:
We can use `COALESCE` instead (with sift) and make it Oracle only just in case.

Reply:
Spoke with Gui, and we discovered that baby squeel does not offer reorder construct yet: https://github.com/rzane/baby_squeel/issues/73

Reply:
Coalesce is not ordering properly for me:
![image](https://user-images.githubusercontent.com/11318903/64605407-57b45e80-d3c4-11e9-98f1-e7fed593db6e.png)

Reply:
`backend_api_configs.reorder('CASE WHEN path IS NULL THEN 1 ELSE 0 END, path DESC')` works for the 3 DBs, however it might be a better idea for speed in the other DBs (not sure, so share your opinions) doing the specific query of Oracle only on Oracle so the others do not execute stuff they do not need, like this:
```ruby
reorder = System::Database.oracle? ? 'CASE WHEN path IS NULL THEN 1 ELSE 0 END, path DESC' : {path: :desc}
backend_api_configs.reorder(reorder)
```

Reply:
in fact then i could make it oracle specific like:
```ruby
reorder = System::Database.oracle? ? 'path DESC NULLS LAST' : {path: :desc}
rules = backend_api_configs.reorder(reorder).each_with_object([]) do |config, rules|
# ...
```

Reply:
So these are the 2 options, and both work in all DBs:
```ruby
rules = backend_api_configs.reorder('CASE WHEN path IS NULL THEN 1 ELSE 0 END, path DESC').each_with_object([]) do |config, rules|
# ...
```

```ruby
reorder = System::Database.oracle? ? 'path DESC NULLS LAST' : {path: :desc}
rules = backend_api_configs.reorder(reorder).each_with_object([]) do |config, rules|
# ...
```

Reply:
Right, good 👍 So I will push the last once 👍 It is also definitely more understandable at first glance 😄

Reply:
Just discovered that `reordering` exists now in baby squeel 😄because i tried it locally after seeing https://www.bountysource.com/issues/53307207-reordering-without-explicit-direction-doesn-t-work-with-last-in-rails-4
So I will do the sifter 😄